### DISCUSSION TITLE
OpenVPN obfuscation

### DISCUSSION THREAD

**Discussion by tkmw1985** [SoftEtherVPN/SoftEtherVPN#1452] [Q&A]
How to set obfuscation on server and client ovpn.conf ?

**Comment by davidebeatrici** [comment]
Different obfuscation modes are available, which one would you like to use?

**Reply by tkmw1985** [reply]
Hello, sir 

For example I use standard openvpn with xor patch xormask or scramble obfuscate in both config server and client. How to do in softwther server config and client openvpn config ?  

And one more question when I build from git last command make -C tmp build error occure make**** tmp no such file or directory !  What cause this error ? How to fix this error ?

**Reply by davidebeatrici** [reply]
> For example I use standard openvpn with xor patch xormask or scramble obfuscate in both config server and client. How to do in softwther server config and client openvpn config ?

You can use the `ProtoOptionsSet` command in `vpncmd` to set protocol-specific options.

https://github.com/SoftEtherVPN/SoftEtherVPN/blob/51a3346f702fdaf3b225483acca10083cc80b624/src/Cedar/Proto_OpenVPN.c#L59-L60

You just have to enable `Obfuscation` and set the correct mask string (`ObfuscationMask`), SoftEther VPN takes care of detecting the mode used by the client.

In the client configuration you're probably looking for:

```
scramble obfuscate myobfuscationmask
```

Ideally `myobfuscationmask` should be replaced with a random string.

> And one more question when I build from git last command make -C tmp build error occure make**** tmp no such file or directory ! What cause this error ? How to fix this error ?

What are the commands you used for the build, specifically?

**Reply by tkmw1985** [reply]
Thank u for quike answer !  Will try again.

**Reply by tkmw1985** [reply]
error while loading shared libraries: libcedar.so: cannot open shared object file: No such file or directory 

After building and installing  error occured. How to fix ? Can not start vpnserver

**Reply by chipitsine** [reply]
if you used "make" and "make install", it is expected to work. if that is not working for you, there most probably something broken.
please describe your steps (step by step), so we can reproduce ourselves and investigate.

**Reply by tkmw1985** [reply]
Build from source code and install

git clone https://github.com/SoftEtherVPN/SoftEtherVPN.git
cd SoftEtherVPN
git submodule init && git submodule update
./configure
make -C build
make -C build install

building and installing good but when  starting vpnserver with command  vpnserver start  error occure every time and i do not know how to fix it !

**Reply by chipitsine** [reply]
what linux distributuve you use?

**Reply by tkmw1985** [reply]
ubuntu 18

**Reply by chipitsine** [reply]
give me 1-2 days

**Reply by chipitsine** [reply]
as a temporary unblocking solution, you can try to find libcedar.so and use LD_LIBRARY_PATH to load a lib from that directory

**Reply by tkmw1985** [reply]
Вопрос можно закрыть так как установить все таки без проблем можно уже готовую сборку deb файла от уважаемого  Dmitry Verkhoturov ! Проблем ни каких не заметил.

**Reply by chipitsine** [reply]
Ну тем не менее, я посмотрю в "make install"

@paskal , тебе тут спасибо прилетело
